File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 4/8/2026, 5:14:26 PM
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        for (int i = 0; i < nums.size(); i++) {
            for (int j = i+1; j < nums.size(); j++) {
                if (nums[i] + nums[j] == target) {
                    return {i,j};
                }
            }
        }
        return {0,0};
    }
};

File: 1-TwoSum/1-TwoSum.py
# Last updated: 4/8/2026, 5:14:25 PM
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        complement = {}
        for i, num in enumerate(nums):
            if num in complement:
                return [i, complement[num]]
            else: 
                complement[target - num] = i
        